openfl/experimental/interface/fl_spec.py

```diff
@@ -264,30 +248,21 @@ def next(self, f, **kwargs):
         if aggregator_to_collaborator(f, parent_func):
             agg_to_collab_ss = self._capture_instance_snapshot(kwargs=kwargs)
 
-            if str(self._runtime) == "FederatedRuntime":
-                if len(FLSpec._clones) == 0:
-                    self.get_clones(kwargs)
-
         # Remove included / excluded attributes from next task
         filter_attributes(self, f, **kwargs)
 
         if str(self._runtime) == "FederatedRuntime":
             if f.collaborator_step and not f.aggregator_step:
                 self._foreach_methods.append(f.__name__)
 
-            if "foreach" in kwargs:
-                self.filter_exclude_include(f, **kwargs)
-                # if "foreach" in kwargs:
-                self.execute_task_args = (
-                    self,
-                    f,
-                    parent_func,
-                    FLSpec._clones,
-                    agg_to_collab_ss,
-                    kwargs,
-                )
-            else:
-                self.execute_task_args = (self, f, parent_func, kwargs)
+            self.execute_task_args = (
+                self,
+                f,
+                parent_func,
+                FLSpec._clones,
+                agg_to_collab_ss,
+                kwargs,
+            )
```

Comment:
If we remove the private attribute filtering from here, then who is responsible for doing it, especially if we execute the FLSpec via the `run()` method?

Reply:
Private attribute filtering is required while transitioning from aggregator to collab (and vice-versa). For `LocalRuntime` it is done inside Participants (openfl/experimental/interface/participants.py) and in `FederatedRuntime` it is done inside Component (openfl/experimental/component/aggregator/aggregator.py and openfl/experimental/component/collaborator/collaborator.py)

Reply:
As a follow-up PR, I'd suggest re-organizing a little bit the code. Right now, everything related to Workflow API resides directly under `openfl/experimental` - instead of `openfl/experimental/workflow` for example. Additionally, under `openfl/experimental/workflow`, it would be great to have separate sub-packages for `local` and `federated` for example. WDYT?